app/models/case_log.rb

```diff
@@ -289,6 +290,16 @@ def self.to_csv
     end
   end
 
+  def soft_min_for_period
+    soft_min = LaRentRange.find_by(start_year: collection_start_year, la:, beds:, lettype:).soft_min
+    "#{soft_value_for_period(soft_min)} #{SUFFIX_FROM_PERIOD[period].presence || 'every week'}"
+  end
+
+  def soft_max_for_period
+    soft_max = LaRentRange.find_by(start_year: collection_start_year, la:, beds:, lettype:).soft_max
+    "#{soft_value_for_period(soft_max)} #{SUFFIX_FROM_PERIOD[period].presence || 'every week'}"
```

Comment:
Does `soft_value_for_period` return anything if no period is set? Does this need to? Or could it skip the whole thing?

Reply:
I think it returns "", maybe worth extracting the check for period before the db call to minimise the # of calls

Reply:
But we have this method `def rent_in_soft_min_range?
    return unless brent && weekly_value(brent) && startdate <...>` which is the conditional for checking the routing. And it returns unless those fields are filled in. We're only calling  `soft_max_for_period` when we load the screen I think, to display the value only

Reply:
Does that mean that by the time we call this we _know_ `period` is set and don't need the `.presence` check here?

Reply:
Yes, we know it's there, but if you mean the presence check here, it's just for getting the suffix. Cause it would be `every week` in most of the cases but it doesn't make sense to write it out into the `SUFFIX_FROM_PERIOD`  constant for every period. I struggled to set a default hash value in a concise way cause we need to freeze our constants and this seemed like the shortest way to do it. So it's not checking for the presence of the period, but rather for the presence of value corresponding to it in the constant

Reply:
Got it